pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -117,6 +124,33 @@ func New(port uint, runtime string) (Interface, error) {
 	return cadvisorClient, nil
 }
 
+func generateCollectorHttpClient(cAdvisorCollectorClientCertFile, cAdvisorCollectorClientPrivateKeyFile string) (*http.Client, error) {
+	//Enable accessing insecure endpoints. We should be able to access metrics from any endpoint
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
@deads2k I think I might be looking at this the wrong way and missing something. Could you please explain why exactly it would be more secure to verify certificates coming from an https metrics endpoint?

Verifying the https endpoint doesn't prevent a pod from getting metrics collected. If they want to get metrics collected, they can just do so by using an http endpoint instead. So this doesn't prevent metrics from being collected by the system or how they system reports custom metrics.

The container itself contains the configuration for custom metrics. If that were to be updated to include its CA it doesn't actually make anything more secure. Since both the CA and the certificate are coming from the pod, the pod is self reporting its credentials which means it cannot be trusted. This has the same level of security as just setting InsecureSkipVerify. I went down this route originally by having cAdvisor read the CA directly from the pod, but its not any more secure to do it this way.

Even with CA support it wont actually work. We get metrics directly from the pod, which means we are accessing the pod via its its ip address. In order for the certificate to be valid, it must be valid for that ip address, which it most likely will not be the case in a lot of situations. Especially for your example above with OpenShift generating the service certificates.